src/validations/auth.validation.js

```diff
@@ -45,19 +43,50 @@ const signupSchema = Joi.object({
     })
 }).options({ abortEarly: false });
 
-export const signupValidator = (req, res, next) => {
-  const userInputValidator = signupSchema.validate(req.body);
-  if (userInputValidator.error) {
+const loginSchema = Joi.object({
+  email: Joi.string()
+    .email({ minDomainSegments: 2 })
+    .required()
+    .trim()
+    .messages({
+      'string.empty': 'Please enter email',
+      'string.email': 'Enter valid email i.e: email@example.com',
+      'any.required': 'Email is required'
+    }),
+  password: Joi.string()
+    .trim()
+    .min(8)
+    .required()
+    .messages({
+      'string.base': 'Invalid type, password must be a string',
+      'string.empty': 'Please enter password',
+      'string.min': 'Password must be atleast 8 characters long',
+      'any.required': 'Password is required'
+    })
+}).options({ abortEarly: false });
+
+
+const validateHandler = (schema, body, res, next) => {
+  const validateUserInput = schema.validate(body);
+  if (validateUserInput.error) {
     const errors = [];
-    for (let i = 0; i < userInputValidator.error.details.length; i += 1) {
-      errors.push(userInputValidator.error.details[i].message.split('"').join(" "));
+    for (let i = 0; i < validateUserInput.error.details.length; i += 1) {
```

Comment:
Do not use `for..loop`. Use `map` or `forEach` instead

Reply:
Ok